### PR TITLE
Migrate documentation for all math functions from scalar_functions.md to code

### LINKED ISSUE #12867: Migrate documentation for all math functions from scalar_functions.md to code
### Is your feature request related to a problem or challenge?

As part of https://github.com/apache/datafusion/issues/12740 this ticket is to migrate all the existing documentation for the above named functions from the static documentation to code and to remove them from the static file. 

Here is the current static content:
* doc source: https://github.com/apache/datafusion/blob/main/docs/source/user-guide/sql/scalar_functions.md
* rendered: https://datafusion.apache.org/user-guide/sql/scalar_functions.html
* Code source: https://github.com/apache/datafusion/tree/main/datafusion/functions/src/math

Need Functions:
(see source)

### Describe the solution you'd like

One (or more) PRs that
1. port the documentation for the above named functions from the static docs to code
2. Bonus points:  add sql examples should be added for all functions
3. Remove the documentation from the static docs
4. Run `./dev/update_function_docs.sh` to update the dynamic documentation

Here are some example PRs of doing this:
1. https://github.com/apache/datafusion/pull/12840 from @jonathanc-n 
1. https://github.com/apache/datafusion/pull/12775 from @Omega359 
3. https://github.com/apache/datafusion/pull/12742

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by juroberttyb**
## Which issue does this PR close?

Closes #12867.

### CODE REVIEW

File: datafusion/functions/src/math/monotonicity.rs [hunk 1]
```diff
@@ -218,14 +545,76 @@ pub fn sqrt_order(input: &[ExprProperties]) -> Result<SortProperties> {
...
+            .with_syntax_example("tanh(numeric_expression)")
+            .with_standard_argument("numeric_expression", "Numeric")
+            .build()
+            .unwrap()
+    })
+}
+
+pub fn get_pow_doc() -> &'static Documentation {
+    static DOCUMENTATION: OnceLock<Documentation> = OnceLock::new();
+
```
Comment (Omega359) [inline datafusion/functions/src/math/monotonicity.rs:610]: Aliases do not need documentation as the script that generates the docs will insert the alias link automatically

**Comment by Omega359** [comment]
Beyond the `pow` removal this PR LGTM 👍

File: datafusion/functions/src/math/monotonicity.rs (hunk 1 above)
Comment (juroberttyb) [reply datafusion/functions/src/math/monotonicity.rs]: No problem, this part is removed. Thank you for taking the time to review the PR!

**Comment by juroberttyb** [comment]
PR updated with `get_pow_doc()` removed.

File: docs/source/user-guide/sql/scalar_functions.md [hunk 2]
```diff
@@ -27,534 +27,6 @@ the rest of the documentation.
 
 [automatically created from the codebase]: https://github.com/apache/datafusion/issues/12740
 
-## Math Functions
```
Comment (alamb) [inline docs/source/user-guide/sql/scalar_functions.md:30]: 😍

**Review by alamb — APPROVED** [review]
Thanks @Omega359  and @juroberttyb  -- this is a great contribution. I started the CI checks on this PR

I'll plan to merge it in when the CI complete

Thank you!

**Comment by Omega359** [comment]
A small cargo fmt fix seems to be required.

**Comment by alamb** [comment]
I took the liberty of merging up from main to resolve a conflict (also by pushing a commit I think this PR's CI will run automatically from now on)

**Comment by alamb** [comment]
Thanks again @juroberttyb  and @Omega359

**Comment by juroberttyb** [comment]
Thank you both for your time and help!

**Pull request merged**